### PR TITLE
Improving Trace Module

### PR DESCRIPTION
**Pull request by nagarev**
## Description

Improves `TraceModuleImpl.java` by applying some sonar suggestions

## Motivation and Context

Code Cleanup

## How Has This Been Tested?

- Unit Tests
- QA

## Types of changes

- [x] Code Improvement
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Requires Activation Code (Hard Fork)

* **Other information**:

N/A

### CODE REVIEW

**Comment by nagarev** [comment]
pipeline:run

**Review by rmoreliovlabs — APPROVED** [review]

**Comment by nagarev** [comment]
pipeline:run

**Comment by nagarev** [comment]
pipeline:run

File: rskj-core/src/main/java/co/rsk/rpc/modules/trace/TraceModuleImpl.java [hunk 1]
```diff
@@ -254,7 +249,7 @@ private List<TransactionTrace> buildBlockTraces(Block block, TraceFilterRequest
 
                 if (programTrace == null) {
                     blockTraces.clear();
-                    return null;
+                    return Collections.emptyList();
```
Comment (illuque-iov) [inline rskj-core/src/main/java/co/rsk/rpc/modules/trace/TraceModuleImpl.java:252]: It was there before, but why clearing the collection if you are just returning a different thing? Just either clear it and `break` or `return Collections.emptyList();`

I would also probably take the chance refactor this method (just a bit) to early exit with an empty collection in line 220 `if (block == null || block.getNumber() == 0)` in order to reduce indentation and ease readability.

Comment (nagarev) [reply rskj-core/src/main/java/co/rsk/rpc/modules/trace/TraceModuleImpl.java:252]: The idea for this PR was to solve only this sonar finding, we can add all those other refactoring changes in another PR for the sake of clarity.

**Review by Vovchyk — APPROVED** [review]

**Pull request merged**